src/common/components/nav/nav.component.js

```diff
@@ -127,7 +129,15 @@ class NavController{
   }
 
   signOut() {
-    this.sessionService.downgradeToGuest().subscribe(angular.noop);
+    let modal = this.$uibModal.open({
+      templateUrl: signOutTemplate.name,
+      backdrop: 'static',
+      keyboard: false,
+      size: 'sm'
+    });
+    this.sessionService.downgradeToGuest().subscribe(() => {
+      modal.close();
+    }, angular.noop);
```

Comment:
Technically, that close() code probably never happens, the page should reload or redirect before it happens. I just added it as a fallback, just in case.